src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderTest.cs

```diff
@@ -225,5 +226,81 @@ private static bool IsColumnBitSet(SqlConnection con, string selectQuery, int in
             }
             return columnSetPresent;
         }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static void CheckHiddenColumns()
+        {
+            // hidden columns can be found by using CommandBehavious.KeyInfo or at the sql level
+            // by using the FOR BROWSE option. These features return the column information requested and
+            // also include any key information required to be able to find the row containing the data
+            // requested. The additional key infomration is provided as hidden columns and can be seen using
+            // the difference between VisibleFieldCount and FieldCount on the reader
+
+            string tempTableName = DataTestUtility.GenerateObjectName();
+
+            string createQuery = $@"
+create table [{tempTableName}] (
+	user_id int not null identity(1,1),
+	first_name varchar(100) null,
+	last_name varchar(100) null);
+
+alter table [{tempTableName}] add constraint pk_{tempTableName}_user_id primary key (user_id);
+
+insert into [{tempTableName}] (first_name,last_name) values ('Joe','Smith')
+";
+
+            string dataQuery = $@"select first_name, last_name from [{tempTableName}]";
+
```

Comment:
Any value in adding a test for a result set with one PK column? Both fieldCount and visibleFieldCount would be 1.

Reply:
If you select a single field that isn't the key you'll get a count of total:visible of 2:1, there isn't a way to select no fields and get the hidden key. The case of 3:2 and 2:1 are the same really so I didn't see any value in adding it. 

You could also say that we should test to see if we can get the hidden field at the start or the middle of the list but I've no way to know how to do that so I didn't try it.

Reply:
Sorry, I meant if you executed `select user_id from [{tempTableName}]`.

Reply:
it'll have the same basic result but the number will be 1 smaller. If you want I can change this one or add a second but it doesn't cover any more functionality to do so. Want me to add or change?

Reply:
I've just started going through the changes but if it doesn't cover any more functionality, then never mind what I said.